cloudinit/config/cc_resizefs.py

```diff
@@ -60,15 +60,28 @@ def _resize_btrfs(mount_point, devpth):
     if not util.mount_is_read_write(mount_point) and os.path.isdir(
         "%s/.snapshots" % mount_point
     ):
-        return (
+        cmd = [
             "btrfs",
             "filesystem",
             "resize",
             "max",
             "%s/.snapshots" % mount_point,
-        )
+        ]
     else:
-        return ("btrfs", "filesystem", "resize", "max", mount_point)
+        cmd = ["btrfs", "filesystem", "resize", "max", mount_point]
+
+    # btrfs has exclusive operations and resize may fail if btrfs is busy
+    # doing one of the operations that prevents resize. As of btrfs 5.10
+    # the resize operation can be queued
+    btrfs_with_queue = util.Version().from_str("5.10")
+    system_btrfs_ver = util.Version().from_str(
+        subp.subp(["btrfs", "--version"])[0].split("v")[-1].strip()
+    )
+    if system_btrfs_ver >= btrfs_with_queue:
```

Comment:
maybe put the messy stuff into a function, `btrfs_ver_gte(str: cmp) -> bool`

Reply:
Sorry what is messy about the `>=` operator?

Reply:
i was thinking that from Transforming the string all the way to the comparison could be put into its own function

Reply:
Well, then `btrfs_ver_gte(str: cmp) -> bool` as a signature would be somewhat confusing IMHO. `btrfs_has_queue_resize() -> bool` would match that idea better. Then again from my perspective the not so nice part is really the indexing into the command list.